### PR TITLE
AWS credential checking

### PR DESCRIPTION
**Pull request by arahayrabedian**
add aws credential checking ONLY FOR YOUR OWN **SECRET** credentials if they're set in a configurable credentials file (AWS CLI tools' format), access keys are ignored.

I haven't had time to figure out how to configure the arguments from a .pre-commit-config yaml file, some help would be nice so I can update the readme further.

Let me know if anything's missing in order for this to get accepted.

### CODE REVIEW

**Comment by asottile** [comment]
Sounds like a great idea!  Write some tests and fix the comments and I think it'll be good for merge

File: pre_commit_hooks/detect_aws_credentials.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import ConfigParser
```
Comment (asottile) [inline pre_commit_hooks/detect_aws_credentials.py]: py2/py3 issue.  Feel free to add `six` as a dependency and `from six.moves import configparser`

File: pre_commit_hooks/detect_aws_credentials.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        keys.add(parser.get(section, 'aws_secret_access_key'))
+    return keys
+
+
+def check_file_for_aws_keys(filename, keys):
+    with open(filename, 'r') as content:
+        # naively match the entire file, chances be so slim
+        # of random characters matching your flipping key.
+        for line in content:
+            if any(key in line for key in keys):
```
Comment (asottile) [inline pre_commit_hooks/detect_aws_credentials.py]:
```python
if key in content.read():
    return 1
```
?

File: pre_commit_hooks/detect_aws_credentials.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        default='~/.aws/credentials',
+        help="location of aws credentials file from which to get the keys "
+              "we're looking for",
+    )
+    parser.add_argument(
+        "--ignore-access-key", 
+        action='store_true',
+        help="if you would like to ignore access keys, as there is "
+        "occasionally legitimate use for these.",
+    )
```
Comment (asottile) [inline pre_commit_hooks/detect_aws_credentials.py]: I think instead they'd either: 1. not use the hook or 2. use `SKIP=detect-aws-credentials git commit ...` (http://pre-commit.com/#advanced)

Comment (arahayrabedian) [inline pre_commit_hooks/detect_aws_credentials.py]: yeah I had seen the option to skip tests, I was thinking about a case where one may want the access key in the code and not the secret but then I realised you'd likely use env vars either way. will drop this option.

File: pre_commit_hooks/detect_aws_credentials.py (hunk 2 above)
Comment (arahayrabedian) [inline pre_commit_hooks/detect_aws_credentials.py]: fair point, don't practice file reading as often as I should :P

File: pre_commit_hooks/detect_aws_credentials.py (hunk 1 above)
Comment (arahayrabedian) [inline pre_commit_hooks/detect_aws_credentials.py]: this one i'll have to play around with, never used six myself but i've seen it all around me. will experiment.

**Comment by arahayrabedian** [comment]
will get around to finishing this off over the weekend i guess.

**Comment by arahayrabedian** [comment]
@asottile i had some issues with pylint and six, couldn't figure it, dunno if it'll happen on the CI server but I was hoping you'd have some pointers on that... all I could find was [this bug](https://bitbucket.org/logilab/pylint/issue/223/ignored-modules-should-turn-no-name-in)

**Comment by asottile** [comment]
@arahayrabedian feel free to `# pylint:disabled= ...` for that file

**Comment by arahayrabedian** [comment]
@asottile anything I need to do to get this in?

**Comment by asottile** [comment]
Oops, sorry I didn't notice your update, I'll review now

**Comment by arahayrabedian** [comment]
no worries, and dankew!

**Comment by asottile** [comment]
Please PR http://github.com/pre-commit/pre-commit.github.io to add your name! <3

**Pull request merged**